great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -136,6 +133,13 @@ class TableAsset(DataAsset):
     column_splitter: Optional[SqlYearMonthSplitter] = None
     name: str
 
+    # non-field private attrs
+    _datasource: SQLDatasource = pydantic.PrivateAttr()
+
+    @property
+    def datasource(self) -> SQLDatasource:
+        return self._datasource
```

Comment:
Why do we need to re-implement this on all the child classes?
Is it just for the better typing?

Reply:
Discussed in meeting today. This is for typing. Hopefully we can find a way around this but it might be in a followup.

Reply:
This was removed after deciding I don't need the child-class type in this PR.